src/application.rs

```diff
@@ -137,20 +176,79 @@ impl Application {
-            let mut dpc = current_dpc.clone();
-
-            dpc.key = "manual".to_string();
-            dpc.time_priority = chrono::Utc::now();
-
-            self.send_ipc_message(IpcMessage::new_request(Request::SetDPC(dpc)));
+    pub fn send_dpc(&mut self, old: InterfaceState, new: InterfaceState) {
+        let current_dpc = self.model.borrow().get_current_dpc().cloned();
+        if let Some(current_dpc) = current_dpc {
+            info!("send_dpc: Sending DPC for iface {}", &new.iface_name);
+            let dpc = current_dpc.clone();
+            let mut new_dpc = dpc.to_new_dpc_with_key("manual");
+            // there are 3 cases:
+            // 1. iface is switched DHCP -> Static
+            // 2. iface is switched Static -> DHCP
+            // 3. iface is switched Static -> Static with different IP
+            //
+            match (old.is_dhcp(), new.is_dhcp()) {
+                (false, true) => {
+                    // case 2
+                    new_dpc
+                        .get_port_by_name_mut(&new.iface_name)
+                        .unwrap()
+                        .to_dhcp();
+                }
+                (_, false) => {
+                    let ip: IpAddr = new.ipv4.parse().unwrap();
+                    let mask: IpAddr = new.mask.parse().unwrap();
+
+                    // parse DNS server string and convert to Option<Vec<IpAddr>>
+                    let dns_servers = new
+                        .dns
+                        .split(',')
+                        .map(|s| s.parse::<IpAddr>().ok())
+                        .flatten()
+                        .collect::<Vec<IpAddr>>();
+
+                    // same for NTP
+                    let ntp_servers = new
+                        .ntp
+                        .split(',')
+                        .map(|s| s.parse::<IpAddr>().ok())
+                        .flatten()
+                        .collect::<Vec<IpAddr>>();
+
+                    // case 1,3
+                    new_dpc
+                        .get_port_by_name_mut(&new.iface_name)
+                        .unwrap()
+                        .to_static(
+                            IpNet::with_netmask(ip, mask).unwrap(),
+                            new.gw.parse().unwrap(),
+                            new.domain,
+                            if ntp_servers.is_empty() {
+                                None
+                            } else {
+                                Some(ntp_servers[0])
+                            },
+                            if dns_servers.is_empty() {
+                                None
+                            } else {
+                                Some(dns_servers)
+                            },
+                        );
+                }
+                _ => {} // do nothing
+            }
+            self.send_ipc_message(IpcMessage::new_request(Request::SetDPC(new_dpc)), |_| {});
```

Comment:
so if nothing changes you still want to send message with "new" dpc, which hasn't been changed?

Reply:
Okay, missed that, you handle it there, makes sense